metadata-ingestion/src/datahub/ingestion/source/pulsar.py

```diff
@@ -78,7 +78,16 @@ class PulsarSchema:
     def __init__(self, schema):
         self.schema_version = schema.get("version")
 
-        avro_schema = json.loads(schema.get("data"))
+        schema_data = schema.get("data")
+        if not schema_data:
+            print(f"[WARNING] Schema data is empty or None. Using default empty schema.")
+            schema_data = "{}"
+
+        try:
+            avro_schema = json.loads(schema_data)
+        except JSONDecodeError as e:
+            avro_schema = {}
```

Comment:
We should have better error reporting for this - we should emit a warning for this

That warning should be part of the source report (e.g. `source.report.warning(...)`). We never use `print` in our codebase

Reply:
yeah，I have modified it once according to the prompt，tks